compiler/qsc_frontend/src/resolve.rs

```diff
@@ -463,20 +510,37 @@ impl Resolver {
         }
     }
 
-    fn bind_open(&mut self, name: &ast::Ident, alias: &Option<Box<ast::Ident>>) {
-        let alias = alias.as_ref().map_or("".into(), |a| Rc::clone(&a.name));
-        if self.globals.namespaces.contains(&name.name) {
+    fn bind_open(&mut self, name: &Idents, alias: &Option<Box<Ident>>) {
+        let Some(id) = self.globals.find_namespace(name.str_iter()) else {
+            self.errors.push(Error::NotFound(
+                name.iter()
+                    .map(|x| x.name.to_string())
+                    .collect::<Vec<_>>()
+                    .join("."),
+                name.span(),
+            ));
+            return;
+        };
+        let alias = alias
+            .as_ref()
+            .map_or(name.into(), |a| vec![Rc::clone(&a.name)]);
+        if self
+            .globals
+            .namespaces
+            .get_namespace_id(name.str_iter())
+            .is_some()
+        {
             self.current_scope_mut()
                 .opens
                 .entry(alias)
                 .or_default()
```

Comment:
What is the purpose of the `or_default()` here? Doesn't that just create a mutable reference to an empty vector that will get thrown away at the end of the statement?

Reply:
I see, so it acts as a kind of find_or_insert.